README.md
# HTML Completion - Extension Visual Studio Code

![Logo HTML Completion](https://code.visualstudio.com/assets/docs/editor/intellisense/intellisense_packagejson.gif)

## Aperçu

HTMLCSS Completion est une extension pour Visual Studio Code qui facilite le développement de fichiers HTML et CSS en fournissant des snippets et des raccourcis pour générer rapidement du code de base et d'autres éléments couramment utilisés.

L'extension comprend un ensemble de snippets pour générer des balises HTML, des styles CSS, des éléments de mise en page, et bien plus encore. Que vous soyez un développeur web débutant ou expérimenté, HTMLCSS Completion peut vous faire gagner du temps en accélérant le processus de développement.

<a href="https://marketplace.visualstudio.com/items?itemName=0KLS0.htmlcss-completion" target="_blank"><img src="./src/assets/Apercu.png"></a>

## Exemples de fonctionnalités

### Génère l'en-tête HTML de base

- Préfixe : `doctype`
- Description : Génère l'en-tête HTML de base, y compris la déclaration doctype, la balise html, et la structure de base.

### Insère un footer

- Préfixe : `foot`
- Description : Insère un pied de page générique.

### Insère un formulaire avec différents champs

- Préfixe : `form`
- Description : Insère un formulaire avec différents champs.

### Insère un lien vers le script jQuery

- Préfixe : `jquery`
- Description : Insère un lien vers le script jQuery.

### A l'aide du raccourci clavier `Ctrl+Shift+P` vous pouvez :

- Préfixe : `Valider le code HTML`
- Description : Ouvre la page de validation W3C pour valider le code HTML.

- Préfixe : `Voir la page de l'extension`
- Description : Ouvre le fichier README.md de l'extension.

- Préfixe : `Liste choix`
- Description : Vous demande de choisir un nombre et intégre le nombre d'élément d'une liste dans votre code HTML.

### Pour voir la liste des snippets disponibles

- Appuyez sur `Ctrl+Space` (Windows, Linux) ou `Cmd+Space` (macOS) pour voir la liste des snippets disponibles dans un fichier
HTML ou CSS.

...

## Utilisation

Pour utiliser les snippets, il suffit de commencer à taper le préfixe dans un fichier HTML et de sélectionner l'option appropriée dans la liste des suggestions.

## Problèmes Connus

Il n'y a actuellement aucun problème connu avec cette extension.

## Notes de Version

Les notes de version sont disponibles dans le fichier CHANGELOG.md.

<b>Version actuelle :</b>

<a href="https://marketplace.visualstudio.com/items?itemName=0KLS0.htmlcss-completion" target="_blank"><img src="https://img.shields.io/badge/Visual%20Studio%20Marketplace-v1.0.7-blue?style=for-the-badge&logo=visual-studio-code" alt="Visual Studio Marketplace Version"></a>

## Support et Commentaires

Si vous rencontrez des problèmes ou avez des commentaires, n'hésitez pas à [signaler un problème](https://github.com/IKLSI/HtmlCSS-Completion/issues) cela m'aidera à améliorer cette extension.

**Profitez de l'extension HTMLCSS Completion pour Visual Studio Code !**

## Auteur

<a href="https://github.com/IKLSI/HtmlCSS-Completion" target="_blank"><img src="https://img.shields.io/badge/GitHub-0KLS0%2FHtmlCSS--Completion-blue?style=for-the-badge&logo=github" alt="GitHub Repository"></a> <a href="https://marketplace.visualstudio.com/items?itemName=0KLS0.htmlcss-completion" target="_blank"><img src="https://img.shields.io/visual-studio-marketplace/d/0KLS0.htmlcss-completion?style=for-the-badge&logo=visual-studio-code" alt="Visual Studio Marketplace Downloads"></a>


CHANGELOG.md
# Change Log

All notable changes to the "html-completion" extension will be documented in this file.

## [1.0.0] - 10/20/2023

### Added

- Initial version of the extension.

## [1.0.1] - 10/21/2023

### Added

- Added a License.
- Added and modified snippets related to CSS.

## [1.0.2] - 10/21/2023

### Added

- Added an extension logo.

## [1.0.3] - 10/22/2023

### Added

- Added different snippets for HTML and CSS.
- Added several executable commands with the keyboard shortcut `Ctrl+Shift+P`.

### Changed

- Improved documentation.

## [1.0.4] - 10/23/2023

### Added

- Added an executable command named `Bloc notes` which opens an additional window in which you can write text,
save it and close it.

### Updated

- Updated the CHANGELOG.md file.
- Updated the README.md file.

## [1.0.5] - 10/23/2023

### Updated

- Updated the LICENSE file.

### Deleted

- Deleted diffents commands becomming useless.

## [1.0.6] - 10/25/2023

### Updated

- Updated the README.md file.
- Updated the structure of the extension.

## [1.0.7] - 10/25/2023

### Updated

- Added the number of downloads of the extension in the README.md file.

src/extension.ts
import * as vscode from 'vscode';

export function activate(context: vscode.ExtensionContext) {
	console.log('Congratulations, your extension "html-completion" is now active!');

	let insertion = vscode.commands.registerCommand('html-completion.insertion', async () => {
		const numItems = await vscode.window.showInputBox({
			prompt: 'Combien d\'éléments <li> voulez-vous créer ?',
			placeHolder: 'Entrez un nombre'
		});

		if (numItems) {
			const indentation = '\t';
			const liItems = Array.from({ length: parseInt(numItems) }, () => `${indentation}<li>$1</li>`).join('\n');
			const editor = vscode.window.activeTextEditor;
			if (editor) {
				editor.insertSnippet(new vscode.SnippetString(`<ul>\n${liItems}\n</ul>`));
			}
		}
	});

	let validateur = vscode.commands.registerCommand('html-completion.validateur', () => {
		vscode.env.openExternal(vscode.Uri.parse('https://validator.w3.org/#validate_by_input'));
	});

	let github = vscode.commands.registerCommand('html-completion.github', () => {
		vscode.env.openExternal(vscode.Uri.parse('https://github.com/IKLSI/HtmlCSS-Completion#html-completion---extension-visual-studio-code'));
		vscode.env.openExternal(vscode.Uri.parse('https://marketplace.visualstudio.com/items?itemName=0KLS0.htmlcss-completion'));
		vscode.window.showInformationMessage('Merci pour votre soutien !');
	});

	let openNotes = vscode.commands.registerCommand('html-completion.openNotesWindow', () => {
		const panel = vscode.window.createWebviewPanel(
			'notesPanel',
			'Bloc-notes',
			vscode.ViewColumn.One,
			{
				enableScripts: true,
			}
		);

		panel.webview.html = getWebViewContent(panel);

		panel.webview.onDidReceiveMessage(message => {
			if (message.command === 'saveNote') {
				const noteContent = message.note;
				vscode.window.showSaveDialog({ filters: { 'Text Files': ['txt'] } }).then(fileUri => {
					if (fileUri) {
						const fs = require('fs');
						fs.writeFileSync(fileUri.fsPath, noteContent);
						vscode.window.showInformationMessage(`Note enregistrée sous ${fileUri.fsPath}`);
					}
				});
			}
		});
	});

	context.subscriptions.push(insertion);
	context.subscriptions.push(validateur);
	context.subscriptions.push(github);
	context.subscriptions.push(openNotes);
}

export function deactivate() {}

function getWebViewContent(panel: vscode.WebviewPanel) {
	const vscodeSettings = vscode.workspace.getConfiguration('workbench');
	const isDarkTheme = vscodeSettings.get('colorTheme') === 'Dark+ (default)';

	return `
	<!DOCTYPE html>
	<html>
	<head>
		<title>Bloc-notes</title>
		<style>
			body {
				background-color: ${isDarkTheme ? '#333' : '#f5f5f5'};
				color: ${isDarkTheme ? 'white' : 'black'};
				margin: 0;
				padding: 0;
				display: flex;
				flex-direction: column;
				align-items: center;
				justify-content: center;
				height: 100vh;
				font-family: poppins;
				font-weight: bold;
			}
	
			h1 {
				font-size: 24px;
				margin-bottom: 20px;
				text-transform: uppercase;
				font-family: arial;
			}
	
			#note-content {
				width: 100%;
				max-width: 600px;
				height: 200px;
				padding: 10px;
				border: 1px solid ${isDarkTheme ? '#777' : '#ccc'};
				border-radius: 5px;
				resize: none;
			}
	
			#save-button {
				background-color: ${isDarkTheme ? '#007ACC' : '#0099E5'};
				color: white;
				border: none;
				padding: 10px 20px;
				border-radius: 5px;
				cursor: pointer;
				font-size: 16px;
				margin-top: 10px;
			}
	
			#save-button:hover {
				background-color: ${isDarkTheme ? '#005F99' : '#007ACC'};
			}
	
			#save-button:active {
				background-color: ${isDarkTheme ? '#004C80' : '#0066B2'};
			}
		</style>
	</head>
	<body>
		<h1>Rédiger une note</h1>
		<textarea id="note-content" rows="10" cols="50"></textarea>
		<br>
		<button id="save-button">Enregistrer</button>
	
		<script>
			const vscode = acquireVsCodeApi();
	
			document.getElementById('save-button').addEventListener('click', () => {
				const noteContent = document.getElementById('note-content').value;
				vscode.postMessage({ command: 'saveNote', note: noteContent });
			});
		</script>
	</body>
	</html>
	`;
}

out/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deactivate = exports.activate = void 0;
const vscode = require("vscode");
function activate(context) {
    console.log('Congratulations, your extension "html-completion" is now active!');
    let insertion = vscode.commands.registerCommand('html-completion.insertion', async () => {
        const numItems = await vscode.window.showInputBox({
            prompt: 'Combien d\'éléments <li> voulez-vous créer ?',
            placeHolder: 'Entrez un nombre'
        });
        if (numItems) {
            const indentation = '\t';
            const liItems = Array.from({ length: parseInt(numItems) }, () => `${indentation}<li>$1</li>`).join('\n');
            const editor = vscode.window.activeTextEditor;
            if (editor) {
                editor.insertSnippet(new vscode.SnippetString(`<ul>\n${liItems}\n</ul>`));
            }
        }
    });
    let validateur = vscode.commands.registerCommand('html-completion.validateur', () => {
        vscode.env.openExternal(vscode.Uri.parse('https://validator.w3.org/#validate_by_input'));
    });
    let github = vscode.commands.registerCommand('html-completion.github', () => {
        vscode.env.openExternal(vscode.Uri.parse('https://github.com/IKLSI/HtmlCSS-Completion#html-completion---extension-visual-studio-code'));
        vscode.env.openExternal(vscode.Uri.parse('https://marketplace.visualstudio.com/items?itemName=0KLS0.htmlcss-completion'));
        vscode.window.showInformationMessage('Merci pour votre soutien !');
    });
    let openNotes = vscode.commands.registerCommand('html-completion.openNotesWindow', () => {
        const panel = vscode.window.createWebviewPanel('notesPanel', 'Bloc-notes', vscode.ViewColumn.One, {
            enableScripts: true,
        });
        panel.webview.html = getWebViewContent(panel);
        panel.webview.onDidReceiveMessage(message => {
            if (message.command === 'saveNote') {
                const noteContent = message.note;
                vscode.window.showSaveDialog({ filters: { 'Text Files': ['txt'] } }).then(fileUri => {
                    if (fileUri) {
                        const fs = require('fs');
                        fs.writeFileSync(fileUri.fsPath, noteContent);
                        vscode.window.showInformationMessage(`Note enregistrée sous ${fileUri.fsPath}`);
                    }
                });
            }
        });
    });
    context.subscriptions.push(insertion);
    context.subscriptions.push(validateur);
    context.subscriptions.push(github);
    context.subscriptions.push(openNotes);
}
exports.activate = activate;
function deactivate() { }
exports.deactivate = deactivate;
function getWebViewContent(panel) {
    const vscodeSettings = vscode.workspace.getConfiguration('workbench');
    const isDarkTheme = vscodeSettings.get('colorTheme') === 'Dark+ (default)';
    return `
	<!DOCTYPE html>
	<html>
	<head>
		<title>Bloc-notes</title>
		<style>
			body {
				background-color: ${isDarkTheme ? '#333' : '#f5f5f5'};
				color: ${isDarkTheme ? 'white' : 'black'};
				margin: 0;
				padding: 0;
				display: flex;
				flex-direction: column;
				align-items: center;
				justify-content: center;
				height: 100vh;
				font-family: poppins;
				font-weight: bold;
			}
	
			h1 {
				font-size: 24px;
				margin-bottom: 20px;
				text-transform: uppercase;
				font-family: arial;
			}
	
			#note-content {
				width: 100%;
				max-width: 600px;
				height: 200px;
				padding: 10px;
				border: 1px solid ${isDarkTheme ? '#777' : '#ccc'};
				border-radius: 5px;
				resize: none;
			}
	
			#save-button {
				background-color: ${isDarkTheme ? '#007ACC' : '#0099E5'};
				color: white;
				border: none;
				padding: 10px 20px;
				border-radius: 5px;
				cursor: pointer;
				font-size: 16px;
				margin-top: 10px;
			}
	
			#save-button:hover {
				background-color: ${isDarkTheme ? '#005F99' : '#007ACC'};
			}
	
			#save-button:active {
				background-color: ${isDarkTheme ? '#004C80' : '#0066B2'};
			}
		</style>
	</head>
	<body>
		<h1>Rédiger une note</h1>
		<textarea id="note-content" rows="10" cols="50"></textarea>
		<br>
		<button id="save-button">Enregistrer</button>
	
		<script>
			const vscode = acquireVsCodeApi();
	
			document.getElementById('save-button').addEventListener('click', () => {
				const noteContent = document.getElementById('note-content').value;
				vscode.postMessage({ command: 'saveNote', note: noteContent });
			});
		</script>
	</body>
	</html>
	`;
}
//# sourceMappingURL=extension.js.map